packages/trader/src/Stores/Modules/Trading/Helpers/process.js

```diff
@@ -11,7 +9,9 @@ export const processTradeParams = async (store, new_state) => {
 
     // To make sure that every function is invoked and affects the snapshot respectively, we have to use for instead of forEach
     for (let i = 0; i < functions.length; i++) {
-        extendOrReplace(snapshot, await functions[i](snapshot)); // eslint-disable-line no-await-in-loop
+        // Shallow copy with Object.assign is good enough to extend the snapshot with new state
+        // we don't need deep extension here, since each function in functions array composes a property of the store completely
+        Object.assign(snapshot, await functions[i](snapshot)); // eslint-disable-line no-await-in-loop
```

Comment:
Is promise.all possible here or do the await function calls need to be in order?

Reply:
await function calls should be in order 😄 
Each function changes some props in snapshot and the snapshot with new props should be passed to next function call.